src/static/helpers/slasHelper.test.ts

```diff
@@ -0,0 +1,263 @@
+/**
+ * @jest-environment node
+ */
+/* eslint header/header: "off" */
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {ShopperLogin, TokenResponse} from '../../lib/shopperLogin';
+import * as slasHelper from './slasHelper';
+
+const codeVerifier = 'code_verifier';
+
+const credentials = {
+  username: 'shopper_user_id',
+  password: 'shopper_password',
+};
+
+const expectedTokenResponse: TokenResponse = {
+  access_token: 'access_token',
+  id_token: 'id_token',
+  refresh_token: 'refresh_token',
+  expires_in: 0,
+  token_type: 'token_type',
+  usid: 'usid',
+  customer_id: 'customer_id',
+  enc_user_id: 'enc_user_id',
+};
+
+const parameters = {
+  redirectURI: 'redirect_uri',
+  refreshToken: 'refresh_token',
+  usid: 'usid',
+};
+
+const url =
+  'https://localhost:3000/callback?usid=048adcfb-aa93-4978-be9e-09cb569fdcb9&code=J2lHm0cgXmnXpwDhjhLoyLJBoUAlBfxDY-AhjqGMC-o';
+
+const authenticateCustomerMock = jest.fn(() => ({url}));
+
+const authorizeCustomerMock = jest.fn(() => ({url}));
+
+const getAccessTokenMock = jest.fn(() => expectedTokenResponse);
+
+const logoutCustomerMock = jest.fn(() => expectedTokenResponse);
+
+const createMockSlasClient = () =>
+  ({
+    clientConfig: {
+      parameters: {
+        organizationId: 'organization_id',
+        clientId: 'client_id',
+        siteId: 'site_id',
+      },
+    },
+    authenticateCustomer: authenticateCustomerMock,
+    authorizeCustomer: authorizeCustomerMock,
+    getAccessToken: getAccessTokenMock,
+    logoutCustomer: logoutCustomerMock,
+  } as unknown as ShopperLogin<{
+    shortCode: string;
+    organizationId: string;
+    clientId: string;
+    siteId: string;
+  }>);
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('Create code verifier', () => {
+  test('creates 128 URL safe string', () => {
```

Comment:
```suggestion
  test('creates 128-character URL-safe string', () => {
```
I hope it doesn't create 128 URLs 🥴